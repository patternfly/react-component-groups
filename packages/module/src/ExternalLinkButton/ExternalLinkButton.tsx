import { Button, ButtonProps } from '@patternfly/react-core';
import { ExternalLinkAltIcon } from '@patternfly/react-icons';
import type { SVGIconProps } from '@patternfly/react-icons/dist/esm/createIcon';
import { forwardRef, Ref } from 'react';

/** extends ButtonProps */
export interface ExternalLinkButtonProps extends ButtonProps {
    /** Additional props to pass to the icon */
    iconProps?: SVGIconProps;
};

export const ExternalLinkButton = forwardRef(({ iconProps, ...props }: ExternalLinkButtonProps, ref: Ref<HTMLAnchorElement>) => (
  <Button
    component="a"
    icon={<ExternalLinkAltIcon {...iconProps} />}
    iconPosition="right"
    ouiaId="ExternalLinkButton"
    ref={ref}
    rel="noopener noreferrer"
    target="_blank"
    {...props}
  />
));

export default ExternalLinkButton;
