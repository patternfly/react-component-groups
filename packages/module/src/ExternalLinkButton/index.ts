export { default } from './ExternalLinkButton';
export * from './ExternalLinkButton';
