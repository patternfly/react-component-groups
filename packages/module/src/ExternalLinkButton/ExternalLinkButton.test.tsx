import { ExternalLinkButton } from './ExternalLinkButton';
import { render } from '@testing-library/react';

describe('ExternalLinkButton component', () => {
  test('should render', () => {
    expect(render(<ExternalLinkButton />)).toMatchSnapshot();
  });

  test('should accept IconProps', () => {
    expect(render(<ExternalLinkButton iconProps={{ className: "my-external-link-icon", title: "(Opens in new tab)" }} />)).toMatchSnapshot();
  });
});
