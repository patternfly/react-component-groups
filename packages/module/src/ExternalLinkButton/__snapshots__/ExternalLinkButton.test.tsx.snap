// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLinkButton component should accept IconProps 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="pf-v6-c-button pf-m-primary"
        data-ouia-component-id="ExternalLinkButton"
        data-ouia-component-type="PF6/Button"
        data-ouia-safe="true"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          class="pf-v6-c-button__icon"
        >
          <svg
            aria-labelledby="icon-title-1"
            class="pf-v6-svg my-external-link-icon"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 512 512"
            width="1em"
          >
            <title
              id="icon-title-1"
            >
              (Opens in new tab)
            </title>
            <path
              d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
            />
          </svg>
        </span>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="pf-v6-c-button pf-m-primary"
      data-ouia-component-id="ExternalLinkButton"
      data-ouia-component-type="PF6/Button"
      data-ouia-safe="true"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        class="pf-v6-c-button__icon"
      >
        <svg
          aria-labelledby="icon-title-1"
          class="pf-v6-svg my-external-link-icon"
          fill="currentColor"
          height="1em"
          role="img"
          viewBox="0 0 512 512"
          width="1em"
        >
          <title
            id="icon-title-1"
          >
            (Opens in new tab)
          </title>
          <path
            d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
          />
        </svg>
      </span>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ExternalLinkButton component should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="pf-v6-c-button pf-m-primary"
        data-ouia-component-id="ExternalLinkButton"
        data-ouia-component-type="PF6/Button"
        data-ouia-safe="true"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          class="pf-v6-c-button__icon"
        >
          <svg
            aria-hidden="true"
            class="pf-v6-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
            />
          </svg>
        </span>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="pf-v6-c-button pf-m-primary"
      data-ouia-component-id="ExternalLinkButton"
      data-ouia-component-type="PF6/Button"
      data-ouia-safe="true"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        class="pf-v6-c-button__icon"
      >
        <svg
          aria-hidden="true"
          class="pf-v6-svg"
          fill="currentColor"
          height="1em"
          role="img"
          viewBox="0 0 512 512"
          width="1em"
        >
          <path
            d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
          />
        </svg>
      </span>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
